jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/ec2/AssetTypeScannedBySourceRule.java

```diff
@@ -0,0 +1,115 @@
+import com.tmobile.pacman.commons.policy.PolicyResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@PacmanPolicy(key = "check-asset-scanned-by-source", desc = "checks for Assets scanned by source", severity = PacmanSdkConstants.SEV_HIGH, category = PacmanSdkConstants.GOVERNANCE)
+public class AssetTypeScannedBySourceRule extends BasePolicy {
+
+    private static final Logger logger = LoggerFactory.getLogger(AssetTypeScannedBySourceRule.class);
+    private static final String POLICY_NAME_FOR_LOGGER = "AssetTypeScannedBySourceRule";
+    private final String VULN_ASSET_LOOKUP_KEY = "asset_lookup_key";
+    private final String VULNERABILITY_INDEX = "vulnerability_index";
+
+    /**
+     * The method will get triggered from Rule Engine with following parameters.
+     *
+     * @param ruleParam          ************* Following are the Rule Parameters********* <br><br>
+     *                           <p>
+     *                           ruleKey : check-asset-scanned-by-source <br><br>
+     *                           <p>
+     *                           target : Enter the target days <br><br>
+     *                           <p>
+     *                           discoveredDaysRange : Enter the discovered days Range <br><br>
+     *                           <p>
+     *                           vulnIndex : Enter the Plugin URL <br><br>
+     * @param resourceAttributes this is a resource in context which needs to be scanned this is provided by execution engine
+     * @return the rule result
+     */
+    @Override
+    public PolicyResult execute(Map<String, String> ruleParam, Map<String, String> resourceAttributes) {
+        logger.debug("{}: execution started", POLICY_NAME_FOR_LOGGER);
+        MDC.put("executionId", ruleParam.get("executionId"));
+        MDC.put("ruleId", ruleParam.get(PacmanSdkConstants.POLICY_ID));
+        String category = ruleParam.get(PacmanRuleConstants.CATEGORY);
+        String severity = ruleParam.get(PacmanRuleConstants.SEVERITY);
+        String vulnerabilityIndex = ruleParam.get(VULNERABILITY_INDEX);
+        String vulnAssetLookupKey = ruleParam.get(VULN_ASSET_LOOKUP_KEY);
+        String vulnerabilitiesEndpoint = PacmanUtils.getPacmanHost(PacmanRuleConstants.ES_URI) + "/" + vulnerabilityIndex + "/_search";
+        if (!PacmanUtils.doesAllHaveValue(category, severity)) {
+            logger.info(PacmanRuleConstants.MISSING_CONFIGURATION);
+            throw new InvalidInputException(PacmanRuleConstants.MISSING_CONFIGURATION);
+        }
+        PolicyResult policyResult = null;
+        if (resourceAttributes != null) {
+            String instanceId = StringUtils.trim(resourceAttributes.get(PacmanRuleConstants.RESOURCE_ID));
+            List<JsonObject> vulnerabilityInfoList = new ArrayList<>();
+            if (instanceId.equals("i-082e542d461714dd2")) {
```

Comment:
You probably meant to remove this debug code

Reply:
yes removed now